3_visualize/src/plot_ecdf.R

```diff
@@ -0,0 +1,136 @@
+#' @param plot_type character string indicating the type of eCDF plot to return.
+#' Options include 1) "all_reaches" to plot one CDF over all reaches; 
+#' 2) "select_reaches" to return one CDF for each reach within `segid_select`; and
+#' 3) "select_times" to return a grid of CDF plots for each month and year in 
+#' `months_select` and `years_select`, respectively. Defaults to "all_reaches".
+#' @param fileout character string with the name of the saved cdf plot, including
+#' the file path and extension. 
+#' @param log_x_axis logical; should the CDF plot include a logged x axis?
+#' Defaults to TRUE.
+#' @param seg_id_select character string indicating which segment identifiers to 
+#' include in the CDF plot. Only used if `plot_type` is "select_reaches".
+#' @param months_select character string or character vector indicating which months
+#' to include in the CDF plots. Format is 2-digit integer months (e.g. "01" for January). 
+#' Only used if `plot_type` is "select_times".
+#' @param years_select character string or character vector indicating which years
+#' to include in the CDF plots. Format is 4-digit integer years (e.g. "2020"). 
+#' Only used if `plot_type` is "select_times".
+#' 
+#' @returns 
+#' Saves png file containing the eCDF plot(s) of model performance. 
+#' 
+plot_ecdf <- function(model_results, plot_type = "all_reaches", fileout, 
+                      log_x_axis = TRUE, plot_points = TRUE,
+                      segids_select, months_select, years_select,
+                      plot_width_in = 5, plot_height_in = 4, 
+                      points_size = 0.5, line_size = 0.25, 
+                      axis_text_size = 10, axis_title_size = 12, 
+                      plot_title = "", plot_title_size = 12, plot_title_face = "plain",
+                      facet_strip_color = "lightgray"){
+  
+  # don't use scientific notation for plot axes
+  options(scipen = 999)
+  
+  # To plot one CDF over all reaches, compute the average RMSE over all dates
+  # for each reach, and then make the CDF for those average values.
+  if(plot_type == "all_reaches"){
+    preds <- model_results %>%
+      group_by(model, PRMS_segid) %>%
+      summarize(rmse = sqrt(mean(errsq)),
+                rmse_log10 = log10(rmse),
+                .groups = "drop")
+    
+    p <- ggplot(preds, aes(x = rmse, color = model)) + 
+      stat_ecdf(geom = "step", pad = FALSE, size = line_size)
+    
+    if(plot_points){
+      p1 <- p + stat_ecdf(geom = "point", pad = FALSE, size = points_size)
+    } else {
+      p1 <- p
+    }
+  }
+
+  # To plot one CDF plot for each reach, compute the RMSE for each date
+  # for each reach, and then make the CDF for those values. (This procedure
+  # is equal to taking the sqrt of errsq for each row of model_results$pred
+  # since each segment has only one error value for every date.)
+  if(plot_type == "select_reaches"){
+    preds <- model_results %>%
+      filter(PRMS_segid %in% segids_select) %>%
```

Comment:
does this make 1 CDF per reach in `segids_select`? Or would we branch over all segids to get a plot for each one?

Reply:
If more than one segment is requested, the function will return a faceted plot where each segment gets its own panel, like this: 
![test_reaches](https://user-images.githubusercontent.com/8785034/206016672-b627e12b-ff06-4c6f-99d1-ed7ba918fa00.png)

This probably gets unwieldy when trying to visualize all of the reaches, so keep me updated with how you're using this function.